geometric_kernels/feature_maps/random_phase.py

```diff
@@ -74,20 +74,26 @@ def __call__(
             lengthscale=params["lengthscale"],
         )
 
+        if is_complex(X):
+            dtype = dtype_complex(params["lengthscale"])
+        else:
+            dtype = B.dtype(params["lengthscale"])
+
         weights = B.power(spectrum, 0.5)  # [L, 1]
 
-        random_phases_b = B.cast(
-            B.dtype(params["lengthscale"]), from_numpy(X, random_phases)
-        )
+        random_phases_b = B.cast(dtype, from_numpy(X, random_phases))
 
         phi_product = self.kernel.eigenfunctions.phi_product(
             X, random_phases_b, **params
         )  # [N, O, L]
 
-        embedding = B.cast(B.dtype(params["lengthscale"]), phi_product)  # [N, O, L]
-        weights_t = B.cast(B.dtype(params["lengthscale"]), B.transpose(weights))
+        embedding = B.cast(dtype, phi_product)  # [N, O, L]
+        weights_t = B.cast(dtype, B.transpose(weights))
```

Comment:
Do we need to cast weights to be complex?

Reply:
We need to cast them into the same type as everything else, so B.cast(dtype,...) is the most intuitive way to do this. But, yes, it is enough to cast them into the corresponding float type.